src/common.jl

```diff
@@ -277,6 +277,46 @@ end
 # `_logpdf` should be implemented and has no default definition
 # _logpdf(d::Distribution{ArrayLikeVariate{N}}, x::AbstractArray{<:Real,N}) where {N}
 
+"""
+    gradlogpdf(d::Distribution{ArrayLikeVariate{N}}, x::AbstractArray{<:Real,N}) where {N}
+
+Evaluate the gradient of the logarithm of the probability density function of `d` at `x`.
+
+This function checks if the size of `x` is compatible with distribution `d`. This check can
+be disabled by using `@inbounds`.
+
+# Implementation
+
+Instead of `gradlogpdf` one should implement `_gradlogpdf(d, x)` which does not have to check the
+size of `x`.
+
+See also: [`pdf`](@ref).
+"""
+@inline function gradlogpdf(
+    d::Distribution{ArrayLikeVariate{N}}, x::AbstractArray{<:Real,M}
+) where {N,M}
```

Comment:
My feeling is that the implicit broadcasting behaviour of `logpdf` is a bit unfortunate and that `logpdf` on multiple variates should be deprecated and instead users should be pointed to `eachvariate` (+ `map` or `broadcast`). Therefore I'm a bit hesitant to add this generalization for `gradlogpdf`.

Reply:
Oh, ok. I was just trying to follow along, but now I understand the TO DO comment to deprecate it. I can remove this part (probably a week from now, after a trip).